src/playercards/cards/NkosiMabati3.ttslua

```diff
@@ -0,0 +1,185 @@
+local chaosBagApi       = require("chaosbag/ChaosBagApi")
+local guidReferenceApi  = require("core/GUIDReferenceApi")
+local playmatApi        = require("playermat/PlaymatApi")
+
+local tokenNames = {
+    "Skull",
+    "Cultist",
+    "Tablet",
+    "Elder Thing",
+    "Auto-fail",
+    "Bless",
+    "Curse",
+    "Frost",
+    "Custom Token"
+}
+
+local dataForToken = {
+    ["Skull"] = { color = "#4A0400E6", icon = "token-skull"},
+    ["Cultist"] = { color = "#173B0BE6", icon = "token-cultist"},
+    ["Tablet"] = { color = "#1D2238E6", icon = "token-tablet"},
+    ["Elder Thing"] = { color = "#4D2331E6", icon = "token-elder-thing"},
+    ["Auto-fail"] = { color = "#9B0004E6", icon = "token-auto-fail"},
+    ["Bless"] = { color = "#9D702CE6", icon = "token-bless"},
+    ["Curse"] = { color = "#633A84E6", icon = "token-curse"},
+    ["Frost"] = { color = "#404450E6", icon = "token-frost"},
+    [""] = { color = "#77674DE6", icon = "token-custom"}
+}
+
+local sigil
+
+function onSave()
+    return JSON.encode(sigil)
+end
+  
+function onLoad(savedData)
+    self.addContextMenuItem("Enable Helper", chooseSigil)
+    sigil= JSON.decode(savedData)
+    if sigil and sigil ~= nil then
+        makeXMLButton(sigil)
+        self.clearContextMenu()
+        self.addContextMenuItem("Clear Helper", deleteButtons)
+    end
+end
+
+function makeXMLButton(chosenToken)
+    self.UI.setXmlTable({
+        {
+        tag="Button",
+        attributes={
+            height=450,
+            width=1400,
+            rotation="0 0 180",
+            scale="0.1 0.1 1",
+            position="0 -55 -22",
+            padding="50 50 50 50",
+            font="font_teutonic-arkham",
+            fontSize=300,
+            iconWidth=400,
+            iconAlignment="Right",
+            onClick="resolveSigil",
+            id=sigil,
+            icon=dataForToken[sigil].icon,
+            color=dataForToken[sigil].color,
+            textColor="White",
+        },
+        value="Resolve",
+        }
+    }
+)
+end 
+
+-- Create dialog window to choose sigil and create sigil-drawing button
+function chooseSigil(playerColor)
+    self.clearContextMenu()
+    self.addContextMenuItem("Clear Helper", deleteButtons)
+    Player[playerColor].showOptionsDialog("Choose Sigil", tokenNames, 1, 
+        function(chosenToken)
+            if chosenToken == "Custom Token" then
```

Comment:
I think it would be nice if you had two small helper functions to convert from "token names in bag" to "human readable token names" (and vice versa). These could be used instead of these "if"-statements that happen in multiple places.